tests/testserver/server.py

```diff
@@ -79,13 +81,33 @@ def _create_socket_and_bind(self):
         sock.listen(0)
         return sock
 
+    def _close_server_sock_ignore_errors(self):
+        try:
+            self.server_sock.close()
+        except IOError:
+            pass
+
     def _handle_requests(self, server_sock):
         for _ in range(self.requests_to_handle):
-            sock = server_sock.accept()[0]
+            sock = self._accept_connection(server_sock)
+            if not sock:
+                break
+
             handler_result = self.handler(sock)
 
             self.handler_results.append(handler_result)
 
+    def _accept_connection(self, server_sock):
```

Comment:
This is an instance method that doesn't use `self` and accepts as a value something stored on `self`, yes?